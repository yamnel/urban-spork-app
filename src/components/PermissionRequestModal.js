import React from 'react';
import {Button, Modal, ModalHeader, ModalBody, ModalFooter} from 'reactstrap';
import {connect} from "react-redux";
// import StaticUserDetail from './StaticUserDetail';
// import EditUserDetail from "./EditUserDetail";
import UrbanSporkAPI from "../api/UrbanSporkAPI";


class PermissionRequestModal extends React.Component {

    // componentWillReceiveProps(nextProps) {
    //     if(nextProps.isOpen){
    //         const userData = UrbanSporkAPI.getUserFullData(nextProps.userId);
    //         userData.then(data => this.setState({userData: data}))
    //     }
    // }
    //
    state = {
        // edit: false
    };


    // handleOnEdit = () => {
    //     this.setState({edit: true})
    // };
    //
    handleOnCancel = () => {
            this.props.toggle();
    };

    handleOnClose = () => {
        // this.setState({edit: false});
        this.props.toggle();
    };

    handleOnGrantReguest = () => {

        let payload = {
            ForId: this.props.requestData.forId,
            ById: this.props.managerId ,
            PermissionsToGrant:{
                [this.props.requestData.permissionId]:{
                    EventType:"GrantPermission",
                    IsPending:true,
                    Reason:"Need Access",
                    RequestedBy:this.props.requestData.byId,
                    RequestedFor:this.props.requestData.forId,
                }
            }

        };
        console.log(payload);
        UrbanSporkAPI.grantPermission(payload).then(()=> {
                this.props.refreshData();
                this.props.refreshDashboard();
        }

        );
        this.props.toggle();

    };

    handleOnDenyReguest = () => {
        let payload = {
        ForId: this.props.requestData.forId,
        ById: this.props.managerId ,
        PermissionsToDeny:{
            [this.props.requestData.permissionId]:{
                EventType:"DenyPermission",
                IsPending:true,
                Reason:"Need Access",
                RequestedBy:this.props.requestData.byId,
                RequestedFor:this.props.requestData.forId,
            }
        }

    };
        console.log(payload);
        UrbanSporkAPI.denyPermission(payload).then(()=> {
                this.props.refreshData();
                this.props.refreshDashboard();
            }

        );
        this.props.toggle();


    };


    render() {
        return (
            <div>
                <Modal isOpen={this.props.isOpen} toggle={this.handleOnClose}>
                    <ModalHeader toggle={this.handleOnClose}>Permission Request</ModalHeader>
                    <ModalBody>
                        {/*If requestData exists do something*/}
                        {this.props.requestData && <p><h5>Access to:</h5> {this.props.requestData.permissionName}</p>}
                        <hr/>
                        {this.props.requestData && <p><h5>Reason:</h5> {this.props.requestData.reason}</p>}
                        {/*{this.state.userData? (this.state.edit? <EditUserDetail  userData={this.state.userData}/>: <StaticUserDetail userData={this.state.userData}/>):null}*/}
                    </ModalBody>
                    <ModalFooter>
                        <Button color="danger" onClick={this.handleOnDenyReguest}>Deny Request</Button>
                        {' '}
                        <Button color="success" onClick={this.handleOnGrantReguest}>Grant Request</Button>
                        {' '}
                        <Button color="secondary" onClick={this.handleOnCancel}>Cancel</Button>
                    </ModalFooter>
                </Modal>
            </div>
        );
    }
}
const mapStateToProps = (state) => {
    return {
        managerId: state.manager.id
    }
};
export default connect(mapStateToProps)(PermissionRequestModal);