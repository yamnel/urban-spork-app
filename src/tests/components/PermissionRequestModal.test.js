import React from 'react';
import {shallow} from 'enzyme';
import PermissionRequestModal from "../../components/PermissionRequestModal";


test('should render PendingRequests component correctly with no data', () => {
    const wrapper = shallow(<PermissionRequestModal/>);
    expect(wrapper).toMatchSnapshot()
});