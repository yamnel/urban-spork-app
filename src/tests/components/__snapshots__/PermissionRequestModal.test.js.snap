// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PendingRequests component correctly with no data 1`] = `
<div>
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransition={
      Object {
        "mountOnEnter": true,
        "timeout": 150,
      }
    }
    fade={true}
    isOpen={false}
    keyboard={true}
    modalTransition={
      Object {
        "timeout": 300,
      }
    }
    onClosed={[Function]}
    onOpened={[Function]}
    role="dialog"
    toggle={[Function]}
    zIndex={1050}
  >
    <ModalHeader
      closeAriaLabel="Close"
      tag="h5"
      toggle={[Function]}
      wrapTag="div"
    >
      Permission Request
    </ModalHeader>
    <ModalBody
      tag="div"
    />
    <ModalFooter
      tag="div"
    >
      <Button
        color="danger"
        onClick={[Function]}
        tag="button"
      >
        Deny Request
      </Button>
       
      <Button
        color="success"
        onClick={[Function]}
        tag="button"
      >
        Grant Request
      </Button>
       
      <Button
        color="secondary"
        onClick={[Function]}
        tag="button"
      >
        Cancel
      </Button>
    </ModalFooter>
  </Modal>
</div>
`;
